var/spack/repos/builtin/packages/intel-parallelstudio/package.py

```diff
@@ -0,0 +1,112 @@
+from spack import *
+import sys, os, re
+
+from spack.pkg.builtin.intel import IntelInstaller, filter_pick
+
+class IntelParallelstudio(IntelInstaller):
+    """Intel Parallel Studio.
+
+    Note: You will have to add the download file to a
+    mirror so that Spack can find it. For instructions on how to set up a
+    mirror, see http://software.llnl.gov/spack/mirrors.html"""
+
+    homepage = "https://software.intel.com/en-us/intel-parallel-studio-xe"
+
+    # TODO: can also try the online installer (will download files on demand)
+    version('composer.2016.2', '1133fb831312eb519f7da897fec223fa',
+            url="file://%s/parallel_studio_xe_2016_composer_edition_update2.tgz" % os.getcwd())
+    version('professional.2016.2', '70be832f2d34c9bf596a5e99d5f2d832',
+            url="file://%s/parallel_studio_xe_2016_update2.tgz" % os.getcwd())
+    version('cluster.2016.2', '70be832f2d34c9bf596a5e99d5f2d832',
+            url="file://%s/parallel_studio_xe_2016_update2.tgz" % os.getcwd())
+    version('composer.2016.3', '3208eeabee951fc27579177b593cefe9',
+            url="file://%s/parallel_studio_xe_2016_composer_edition_update3.tgz" % os.getcwd())
+    version('professional.2016.3', 'eda19bb0d0d19709197ede58f13443f3',
+            url="file://%s/parallel_studio_xe_2016_update3.tgz" % os.getcwd())
+    version('cluster.2016.3', 'eda19bb0d0d19709197ede58f13443f3',
+            url="file://%s/parallel_studio_xe_2016_update3.tgz" % os.getcwd())
+
+    variant('rpath', default=True, description="Add rpath to .cfg files")
+    variant('all', default=False, description="Install all files associated with the requested edition")
+    variant('mpi', default=True, description="Install the Intel MPI library and ITAC tool")
+    variant('mkl', default=True, description="Install the Intel MKL library")
+    variant('daal', default=True, description="Install the Intel DAAL libraries")
+    variant('ipp', default=True, description="Install the Intel IPP libraries")
+    variant('tools', default=True, description="Install the Intel Advisor, VTune Amplifier, and Inspector tools")
+
+    provides('mpi', when='@cluster:+mpi')
+    provides('mkl', when='+mkl')
+    provides('daal', when='+daal')
+    provides('ipp', when='+ipp')
+
+    def install(self, spec, prefix):
+
+        # remove the installation DB, otherwise it will try to install into location of other Intel builds
+        try:
+            os.remove(os.path.join(os.environ["HOME"], "intel", "intel_sdp_products.db"))
+        except OSError:
+            pass # if the file does not exist
```

Comment:
Can you change this to an if-statement? There's been conversations about try/except vs if-statements elsewhere (see https://github.com/LLNL/spack/pull/848#discussion_r61271503).